### ISSUE TITLE
Faculty numbers

### ISSUE THREAD

**Issue by jnfoster** [krixly/krixly.github.io#2]
Scraping faculty data from CSRankings is convenient, but it doesn't seem quite right for comparing teaching load. In particular, CSRankings focuses on publications in CS venues and includes faculty with primary appointments in adjacent departments. The issue is that at most institutions, teaching is homed in departments, so those faculty are not contributing to undergraduate teaching.

For instance, at Cornell, by my count there are 36 faculty with tenure-track appointments in CS who participate in undergraduate teaching in Ithaca (and that includes people with administrative appointments like the Dean, Chair, etc.) There are an additional 9 faculty with appointments at Cornell Tech in NYC (where there are no undergrads.) However, CS Rankings, lists 74 faculty at Cornell, which includes people with primary appointments in a range of other fields -- e.g., ECE, IS, etc.

**Comment by krixly** [comment]
I agree - we had some issues with figuring out the number of faculty at each school and this was the compromise we settled on. In addition, our project does not take into account lecturers who teach. We thought about scraping identities from course schedules, but ran into two main issues. One was figuring out how many semesters (or quarters, or trimesters) of data we should look at (as not every professor/lecturer teaches every semester), the second one was that the information did not seem to be readily available, for the same period of time, for every school. We also thought about scraping information from the department websites, but were warned that they were often not updated when faculty leave.

Do you have any recommendations for finding accurate numbers, besides sending out inquiries to members of the departments? That is an option we are heavily considering, because we believe it should be the most accurate.

**Comment by jnfoster** [comment]
Yeah, I don't know of a way besides asking department chairs to tally their "teaching-active" faculty in CS for 2017. Maybe with just 20 such queries it wouldn't be hard to get this data?
